From cd508653043e6e5ee24654746fa5af4e3d0ed8b5 Mon Sep 17 00:00:00 2001
From: "Gu, Wangyi" <wangyi.gu@intel.com>
Date: Tue, 15 Jan 2013 10:01:22 +0800
Subject: [PATCH] [PORT FROM MAIN] Movie studio: fix a memory leak issue in the
 AOSP code

It's an android defect, connect()/disconnect() of OMXClient need be called in a pair,
if miss calling disconnect() while connect() is called, a memory leak will happen.

Change-Id: I91f6c6af9b6567b4af73fe4cfbed118edcaaccbd
Category: aosp improvement
Domain: Video.Media-video editor
Origin: internal
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>

diff --git a/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp b/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
index 5386cdd..93f3cf3 100755
--- a/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
+++ b/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
@@ -779,6 +779,7 @@ M4OSA_ERR queryVideoDecoderCapabilities
     logSupportDecodersAndCapabilities(pDecoders);
     *decoders = pDecoders;
 cleanUp:
+    client.disconnect();
     return err;
 }
 /********************
